node/full.go

```diff
@@ -187,12 +167,80 @@ func newFullNode(
 	}
 
 	node.BaseService = *service.NewBaseService(logger, "Node", node)
-
-	node.P2P.SetTxValidator(node.newTxValidator())
+	node.p2pClient.SetTxValidator(node.newTxValidator())
 
 	return node, nil
 }
 
+func initProxyApp(clientCreator proxy.ClientCreator, logger log.Logger) (proxy.AppConns, error) {
+	proxyApp := proxy.NewAppConns(clientCreator, proxy.NopMetrics())
+	proxyApp.SetLogger(logger.With("module", "proxy"))
+	if err := proxyApp.Start(); err != nil {
+		return nil, fmt.Errorf("error starting proxy app connections: %v", err)
```

Comment:
Might be worth wrapp this error.

Reply:
Using `fmt.Errorf` is equivalent to wrapping.